File: leetcode/medium/167. Two Sum II - Input Array Is Sorted/python/twoSumII.py
"Two Sum II - Input Array is Sorted"
# My Approach
def two_sum_II(numbers, target):
    right_indexes = []
    for operand_pointer_1 in range(len(numbers) - 1):
        operand_pointer_2 = operand_pointer_1 + 1
        if numbers[operand_pointer_1] == numbers[operand_pointer_2] and numbers[operand_pointer_1] + numbers[operand_pointer_2] != target:
            continue
        while operand_pointer_2 < len(numbers):
            result = numbers[operand_pointer_1] + numbers[operand_pointer_2]
            if result == target:
                right_indexes.append(operand_pointer_1 + 1)
                right_indexes.append(operand_pointer_2 + 1)
                return right_indexes
            elif result > target:
                break
            operand_pointer_2 += 1

# Testing
n0 = [2,7,11,15]
t0 = 9
n1 = [2,3,4]
t1 = 6
n2 = [-1, 0]
t2 = -1
n3 = [5,25,75]
t3 = 100
n4 = [3,24,50,79,88,150,345]
t4 = 200
n5 = [-1 for x in range(198)]
n5.append(1)
n5.append(1)
t5 = 2
n6 = [-1,-1] + [1 for x in range(198)]
t6 = -2
print(two_sum_II(n6, t6))

'''Complexity Analysis:
Time Complexity: O(N)
Space Complexity: O(1)'''

# Approach 1: Two Pointers
def twoSumII(numbers, target):
    left_pointer = 0
    right_pointer = len(numbers) - 1
    sum_indexes = [-1, -1]
    while left_pointer < right_pointer:
        result = numbers[left_pointer] + numbers[right_pointer]
        if result == target:
            sum_indexes = [left_pointer + 1, right_pointer + 1]
            return sum_indexes
        elif result > target:
            right_pointer -= 1
        else:
            left_pointer += 1
    return sum_indexes

File: leetcode/medium/167. Two Sum II - Input Array Is Sorted/js/twoSumII.js
"use strict";
// Two Sum II
/**
 * @param {Array int} numbers
 * @param {int} target
 * @return {Array int}
 * @author {Alex Soto}
 */

// Approach 1: Two Pointers
var twoSumII = function(numbers, target) {
    let leftPointer = 0;
    let rightPointer = numbers.length - 1;
    while (leftPointer < rightPointer) {
        let result = numbers[leftPointer] + numbers[rightPointer];
        if (result === target) {
            let sumIndexes = [];
            sumIndexes.push(leftPointer + 1);
            sumIndexes.push(rightPointer + 1);
            return sumIndexes;
        } else if (result > target) {
            rightPointer--;
        } else {
            leftPointer++;
        };
    };
    return [-1, -1]
};

// Testing
let n0 = [2,7,11,15]
let t0 = 9
let n1 = [2,3,4]
let t1 = 6
let n2 = [-1, 0]
let t2 = -1
let n3 = [5,25,75]
let t3 = 100
let n4 = [3,24,50,79,88,150,345]
let t4 = 200
let n5 = [];
for (let i = 0; i <= 197; i++) {
    n5.push(-1);
};
n5.push(1);
n5.push(1);
// console.log(n5);
let t5 = 2;
console.log(twoSumII(n5, t5));

// Complexity Analysis:
// Time Complexity: O(N)
// Space Complecity: O(1)